.changes/unreleased/Refactor-20240710-105829.yaml

```diff
@@ -0,0 +1,3 @@
+kind: Refactor
+body: improve OptionalStringListValue(), dropped context input arg and diagnostics ouput arg
```

Comment:
This seems like a fully internal change and likely won't require a customer facing change entry

Reply:
True, I can delete this